### ISSUE TITLE
Not getting any reading from raspberry pi serial

### ISSUE THREAD

**Issue by l1c2d3** [deiger/Alarm#12]
Hi @deiger ,

I followed the instructions and I´m not getting any reading via serial, not even garbage.
Is there an special firmware version required? 
My panel is a PIMA HUNTER 896 with firmware version HP896-ES-es Rev. 6.08

Given the voltage difference between the raspberry pi and the tx/rx pins from the panel, I even tried with a logic shift converter circuit, but still the same luck.

I even tried [yeudas](https://github.com/yeudas/PimaEspHome) solution based on your work using an esp8266 instead of a raspberry pi, but I couldn't get any serial reading with it.

Any help would be much appreciated.

Thanks in advance.

**Comment by Jens-Wymeersch** [comment]
@l1c2d3 did you ever get it to work ? If so, could you please share what you have done ?

**Comment by l1c2d3** [comment]
> @l1c2d3 did you ever get it to work ? If so, could you please share what you have done ? I have also a HP896

No, I couldn't figure it out, so I abandoned it for the time being.
I came to conclusion that it is firmware related (I have an old version), but being unable to try another version, since I haven't found an updated firmware version chip online.

Hope you have better luck, and if so, please share. So will I, once I get back to work on that project. I will hook up my computer with a serial adapter.

**Comment by Jens-Wymeersch** [comment]
@l1c2d3 I talked to PIMA last week, and they told me that HP896 was too old and most probably wouldn't work. I've now noticed that in the documentation of the PIMA Home Automation Kit, the version of the controller is not RXN410 instead of RXN400 which I have. My assumption is that in order for this to work, I need to upgrade. 
I'm going to call them again to examine if my assumption is correct and see how much this would cost. I'll keep you posted. Please also have a look from your side, we might be able to negotiate if we join forces. Living in Modi'in by the way.

**Comment by maorcc** [comment]
I have a PI Zero connected directly to the PIMA board serial connector.  Works fine for a long time now.

**Comment by Jens-Wymeersch** [comment]
@maorcc thank you for sharing your wiring. I'll give it another go today. 
Another few questions :
- how did the command look like that you used ?
- which Hunter Pro do you have ? an HP896 ?

**Comment by Jens-Wymeersch** [comment]
@maorcc the command i just looks like `python3 ./pima_server.py -p 7777 -k b0c6f700-6098-11eb-ae93-0242ac130002 -l 000000 -z 96 --serialport /dev/serial0 --mqtt_host XX.YY.ZZ.UU`
but somehow it doesn't work

**Comment by Jens-Wymeersch** [comment]
@maorcc The test command looks something like :  `python3 ./pima_server.py --port 7777 --key 28e6db84-44a4-410b-8f58-8c2275fdd396 --zones 96 --login 000000 --serialport /dev/serial0 --mqtt_host XX.YY.ZZ.UU` 
but it doesn't do anything. Can you please get your take on this ?

**Comment by deiger** [comment]
@Jens-Wymeersch maybe it is a different serial port?
Could you try listing the ports, by `ls -l /dev/serial/by-id`?

**Comment by Jens-Wymeersch** [comment]
@deiger 
I'm getting the following
![image](https://user-images.githubusercontent.com/70786498/106442607-24c0ae00-6484-11eb-9098-0ba4124a2a5d.png)

**Comment by deiger** [comment]
According to all the documentation I could find on the web, you should be connecting pins 6, 8, 10.
Maybe you need to enable the UART? See "Disable Linux serial console" [here](https://www.raspberrypi.org/documentation/configuration/uart.md).

**Comment by Jens-Wymeersch** [comment]
@deiger I've enabled the UART as described in the document you send. 
I've connected the cables to the USB to TTL adapter and get the following (setting it at 3.3V instead of 5V)

![image](https://user-images.githubusercontent.com/70786498/106507973-76912480-64d4-11eb-9da3-64d2a0d4c523.png)

I used then the command as follows : 
`python3 ./pima_server.py --port 7777 --key 28e6db84-44a4-410b-8f58-8c2275fdd396 --zones 96 --login 000000 --serialport /dev/ttyUSB0 --mqtt_host XX.YY.ZZ.UU --log_level DEBUG`

The program starts but in the browser on XX.YY.ZZ.UU:7777 nothing happens nor do I see anything on MQTT. 

Here is what you can see in the log
`
Feb  1 21:41:35 raspberrypi D0201 21:41:35.082  pima_server.py:65] Port: /dev/ttyUSB0.
Feb  1 21:42:39 raspberrypi dockerd[557]: time="2021-02-01T21:42:39.734881553+02:00" level=info msg="ignoring event" container=bae1c760dd2d4f744e34994d7fd58d9:`

**Comment by deiger** [comment]
You don't need to connect the power pin at all. It serves to power the connected device, but the Alarm has a separate power source.
Try running it in docker:
1. Create the following config file in `/opt/pima/options.json`
```json
{
  "log_level": "INFO",
  "login": "000000",
  "zones": "96",
  "mqtt_discovery_max_zone": 30,
  "mqtt_host": "XX.YY.ZZ.UU",
  "key": "28e6db84-44a4-410b-8f58-8c2275fdd396",
  "port": 7777,
  "mqtt_user": "homeassistant",
  "mqtt_pass": "password",
  "serialport": "/dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A50295BI-if00-port0"
}
```
1. Run:
```bash
sudo docker run -d --name pima --log-driver journald --restart always -v /opt/pima:/data deiger/pima:0.7.1.6
```
The logs will be available in:
```bash
journalctl CONTAINER_NAME=pima
```

**Comment by Jens-Wymeersch** [comment]
Hi Dror,

Thank you for your quick reply. I got the container working with no errors initially but within the running container, it still have trouble with the serial connection it seems.

Here you have the error message in Portainer :

```
Traceback (most recent call last):,
  File "pima_server.py", line 85, in __init__,
    self._create_alarm(),
  File "pima_server.py", line 139, in _create_alarm,
    self._alarm = pima.Alarm(*self._alarm_args)  # type: pima.Alarm,
  File "/app/pima.py", line 146, in __init__,
    raise Error('Failed to connect to serial port.') from e,
pima.Error: Failed to connect to serial port.,
E0202 13:31:58.396  pima_server.py:87] Failed to create alarm object.,
Traceback (most recent call last):,
  File "/usr/local/lib/python3.7/site-packages/pyserial-3.5-py3.7.egg/serial/serialposix.py", line 322, in open,
    self.fd = os.open(self.portstr, os.O_RDWR | os.O_NOCTTY | os.O_NONBLOCK),
FileNotFoundError: [Errno 2] No such file or directory: '/dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A50285BI-if00-port0',
During handling of the above exception, another exception occurred:,
Traceback (most recent call last):,
  File "/app/pima.py", line 143, in __init__,
    timeout=1),
  File "/usr/local/lib/python3.7/site-packages/pyserial-3.5-py3.7.egg/serial/serialutil.py", line 244, in __init__,
    self.open(),
  File "/usr/local/lib/python3.7/site-packages/pyserial-3.5-py3.7.egg/serial/serialposix.py", line 325, in open,
    raise SerialException(msg.errno, "could not open port {}: {}".format(self._port, msg)),
serial.serialutil.SerialException: [Errno 2] could not open port /dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A50285BI-if00-port0: [Errno 2] No such file or directory: '/dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A50285BI-if00-port0',
The above exception was the direct cause of the following exception:,
```

And here you find the error output from journalctl

Feb 02 15:19:39 raspberrypi 9826ee42b554[557]: During handling of the above exception, another excep
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]: Traceback (most recent call last):
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:   File "/app/pima.py", line 143, in __init__
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:     timeout=1)
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:   File "/usr/local/lib/python3.7/site-packages/pyseri
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:     self.open()
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:   File "/usr/local/lib/python3.7/site-packages/pyseri
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:     raise SerialException(msg.errno, "could not open
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]: serial.serialutil.SerialException: [Errno 2] could no
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]: The above exception was the direct cause of the follo
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]: Traceback (most recent call last):
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:   File "pima_server.py", line 85, in __init__
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:     self._create_alarm()
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:   File "pima_server.py", line 139, in _create_alarm
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:     self._alarm = pima.Alarm(*self._alarm_args)  # ty
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:   File "/app/pima.py", line 146, in __init__
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]:     raise Error('Failed to connect to serial port.')
Feb 02 15:19:39 raspberrypi 9826ee42b554[557]: pima.Error: Failed to connect to serial port.
Feb 02 15:20:41 raspberrypi 9826ee42b554[557]: E0202 13:20:41.243  pima_server.py:87] Failed to crea
Feb 02 15:20:41 raspberrypi 9826ee42b554[557]: Traceback (most recent call last):
Feb 02 15:20:41 raspberrypi 9826ee42b554[557]:   File "/usr/local/lib/python3.7/site-packages/pyseri
Feb 02 15:20:41 raspberrypi 9826ee42b554[557]:     self.fd = os.open(self.portstr, os.O_RDWR | os.O_
Feb 02 15:20:41 raspberrypi 9826ee42b554[557]: FileNotFoundError: [Errno 2] No such file or director

**Comment by deiger** [comment]
Right, we need to map the serial port as well:
```bash
sudo docker run -d --name pima --log-driver journald --restart always -v /opt/pima:/data --device /dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A50295BI-if00-port0:/dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A50295BI-if00-port0 deiger/pima:0.7.1.6
```

**Comment by deiger** [comment]
Alternatively you can do:
```
sudo docker run -d --name pima --log-driver journald --restart always -v /opt/pima:/data --device /dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A50295BI-if00-port0:/dev/serial deiger/pima:0.7.1.6
```

And then set the port in the options file to `/dev/serial`.

**Comment by Jens-Wymeersch** [comment]
> sudo docker run -d --name pima --log-driver journald --restart always -v /opt/pima:/data --device /dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A50295BI-if00-port0:/dev/serial deiger/pima:0.7.1.6

@deiger same result - no errors any more but also no entry in MQTT.

**Comment by deiger** [comment]
Hmm...
Try changing the `log_level` in `options.json` to `DEBUG`.
It seems strange that you won't see anything in journalctl.

**Comment by Jens-Wymeersch** [comment]
@deiger 
this is what I get 
`Feb 02 21:29:02 raspberrypi 376990f53ad7[557]: D0202 19:29:02.900  pima_server.py:70] Port: /dev/serial.`

**Comment by deiger** [comment]
Are you sure the serial port is enabled in the Alarm? You are supposed to have a lot of data flowing in.

**Comment by Jens-Wymeersch** [comment]
@deiger I toggled on the controller the first L from - to + and entr'ed. I moved the tx and rx cable and got this 

`D0202 20:16:19.262  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:20.283  pima.py:265] >>> 62 0d 05 01 02 00 00 80 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 06 0b,
D0202 20:16:20.284  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:21.560  pima.py:265] >>> 62 0d 05 01 02 00 00 80 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 06 0b,
D0202 20:16:21.561  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:22.597  pima.py:265] >>> 62 0d 05 01 02 00 00 80 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 06 0b,
D0202 20:16:22.598  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:23.619  pima.py:265] >>> 62 0d 05 01 02 00 00 80 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 06 0b,
D0202 20:16:23.619  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:24.640  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:24.641  pima.py:284] <<< 04 0d 05 00 00 6d e3,
I0202 20:16:24.642  pima_server.py:135] Status: {'logged in': True, 'open zones': set(), 'alarmed zones': set(), 'bypassed zones': set(), 'failed zones': set(), 'partitions': {1: 'disarm', 2: 'disarm', 3: 'disarm', 4: 'disarm', 5: 'disarm', 6: 'disarm', 7: 'disarm', 8: 'disarm', 9: 'disarm', 10: 'disarm', 11: 'disarm', 12: 'disarm', 13: 'disarm', 14: 'disarm', 15: 'disarm', 16: 'disarm'}, 'command ack': False}.,
D0202 20:16:25.661  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:25.662  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:27.959  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:27.960  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:28.981  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:28.981  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:30.002  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:30.003  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:31.039  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:31.040  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:32.364  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:32.365  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:33.402  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:33.402  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:34.423  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:34.424  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:35.444  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:35.445  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:36.448  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:36.449  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:37.695  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:37.695  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:38.716  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:38.716  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:39.753  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:39.754  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:40.759  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:40.759  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:41.796  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:41.797  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:43.073  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:43.074  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:44.094  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:44.095  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:45.115  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:45.116  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:46.137  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:46.138  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:47.174  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:47.175  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:48.435  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:48.436  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:49.472  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:49.473  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:50.493  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:50.494  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:51.515  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:51.516  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:52.536  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:52.537  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:53.892  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:53.893  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:54.898  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:54.899  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:55.919  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:55.920  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:56.956  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:56.957  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:57.961  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:57.962  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:16:59.190  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:16:59.191  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:17:00.228  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:17:00.228  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:17:01.265  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:17:01.266  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:17:02.270  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:17:02.271  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:17:03.292  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:17:03.292  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:17:04.568  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:17:04.569  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:17:05.606  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:17:05.606  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:17:06.627  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:17:06.628  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:17:07.648  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:17:07.649  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:17:08.670  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:17:08.671  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:17:09.946  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:17:09.947  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:17:10.968  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:17:10.968  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:17:11.989  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:17:11.989  pima.py:284] <<< 04 0d 05 00 00 6d e3,
D0202 20:17:13.026  pima.py:265] >>> 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 01 66 55,
D0202 20:17:13.026  pima.py:284] <<< 04 0d 05 00 00 6d e3,`

**Comment by deiger** [comment]
RX should be connected to TX and vice versa, I believe that is stated in the diagram.
If it works, then that's it. Change the `log_level` back to `INFO`, and see that your alarm appears in HA.
The docker command should maintain it alive through restarts.

**Comment by Jens-Wymeersch** [comment]
@deiger what do you mean with ```
Change the log_level back to INFO, and see that your alarm appears in HA.
The docker command should maintain it alive through restarts.
```
Should I change the options.json file, delete the current docker image/container and re-build them ?
PS I see already see an entry in HA/mosquitto

**Comment by deiger** [comment]
You don't need to delete the container. Just update the config file, and restart the container, so it will pick up the new value.

**Comment by Jens-Wymeersch** [comment]
@deiger Thank you for all the work. I got the items in HA. A few remaining questions 
I got the following card : 
![image](https://user-images.githubusercontent.com/70786498/106727118-d17d6580-6613-11eb-8335-b3609aefa510.png)
- what is the difference btw Arm home and Arm away ?
- does it arm all zones or not ?
- if yes, would it be possible to arm only one zone ? If so, how ?

**Comment by Jens-Wymeersch** [comment]
@deiger when I put ARM AWAY - it sets off the alarm (sirene is heard, etc.)

**Comment by deiger** [comment]
Arm Away does full arm, like a long press on 1.
Arm Home arms home 1, like a long press on 4.

**Comment by Jens-Wymeersch** [comment]
@deiger we typically use 7 to arm our house. I suppose commands 1 and 4 is something set by the alarm company. Right ?

**Comment by deiger** [comment]
7 is Home 2. The server supports it, but HA supports only one Arm Home, and since typically Home 1 is used and Home 2 isn't, I mapped Arm Home to Home 1.
You can either change the mapping code, or change your alarm's configuration. Since you have the Technician code, you can do the latter.

**Comment by Jens-Wymeersch** [comment]
thanks @deiger 
Something happened in the meanwhile. I see the following message on MQTT 
`1612347775: Client auto-7F398332-7F0C-F910-45C7-C4C96CD4AB33 has exceeded timeout, disconnecting.`
Any reason why this happened and how I can avoid it for the future ?

**Comment by deiger** [comment]
Not sure, anything appeared in parallel in the logs of the container?

**Comment by Jens-Wymeersch** [comment]
@deiger Since then everything has run as expected. I've restarted Mosquitto and the container. Thank you again for all the hard work and helping through the hoops.

**Comment by Jens-Wymeersch** [comment]
@deiger it's been running stable since. One last question, if during a break-in the alarm goes off, will I somehow be alerted as well via the system. If so, how ?

**Comment by deiger** [comment]
There are two effects to an alarm being set off:
1. The relevant sensor starts alarming.
1. In the recent version (0.7.2.3) the outputs for the external and internal siren (1 and 2) will be set, and this would result in a state change to "triggered".

**Comment by Jens-Wymeersch** [comment]
@deiger As I've just installed the system, I assume that I have the latest version correct ?

**Comment by deiger** [comment]
The last update was yesterday, so that depends.
If you're using the HA add-on, it should show the current version at the top of the add on page.
You can also see all the version by running:
```
docker ps -a
```

**Comment by Jens-Wymeersch** [comment]
I'm running it seems deiger/pima:0.7.1.6. So I guess I'll need to remove the current version and redo the installation procedure for the next. Right ?

**Comment by l1c2d3** [comment]
@Jens-Wymeersch sorry, not writting back to you, I fell ill with covid. Fully recovered now. 
Glad, you had the help you needed.
I got back to work today with the raspberry  pi and the alarm, and my problem was, that I forgot to re-enable uart after formatting the sd card. The help you got, made me realized that today.

@deiger My motion sensors wired and wireless are not updating (state is always off), do they have to be configured in a different way than a door/window sensor?
Door/window sensors work fine.
I have 1 wired motion sensor, not updating. I have some other connected via pima wireless expander, and I added them manually in HA, not updating either.

(This is an example of one of my motion sensors)
  - platform: mqtt
    name: "Example Zone Movement Detected"
    unique_id: pima_alarm_zone_20_open
    device:
      identifiers:
        - pima_alarm
      manufacturer: PIMA
      model: Hunter Pro 896
      name: PIMA Alarm
      via_device: pima_alarm
    state_topic: pima_alarm/status
    availability_topic: pima_alarm/LWT
    payload_on: 'on'
    payload_off: 'off'
    value_template: '{% if 20 in value_json[''open zones''] %}on{% else %}off{% endif %}'

**Comment by Jens-Wymeersch** [comment]
@deiger I've reinstalled the docker to the new version following the same commands as previously but get now following error message in portainer 
`Traceback (most recent call last):,
E0207 07:36:06.484  pima_server.py:87] Failed to create alarm object.,
Traceback (most recent call last):,
  File "/app/pima.py", line 145, in __init__,
    timeout=1),
  File "/usr/local/lib/python3.7/site-packages/pyserial-3.5-py3.7.egg/serial/serialutil.py", line 244, in __init__,
    self.open(),
  File "/usr/local/lib/python3.7/site-packages/pyserial-3.5-py3.7.egg/serial/serialposix.py", line 325, in open,
    raise SerialException(msg.errno, "could not open port {}: {}".format(self._port, msg)),
serial.serialutil.SerialException: [Errno 21] could not open port /dev/serial: [Errno 21] Is a directory: '/dev/serial'`

**Comment by deiger** [comment]
@Jens-Wymeersch are you mapping `/dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A50295BI-if00-port0`, per [this](https://github.com/deiger/Alarm/issues/12#issuecomment-771704470)?
Do you see `usb-FTDI_FT232R_USB_UART_A50295BI-if00-port0` in `ls -l /dev/serial/by-id`?

**Comment by deiger** [comment]
@l1c2d3 they work exactly the same. All the currently live sensors should appear in `pima_alarm/status`, under "open zones". Note that it takes a few seconds for the sensor to update its status (due to hardware limitations).
One thing that I did find useful is to customize the sensors, setting a different `device_class` in `customize.yaml`.

**Comment by Jens-Wymeersch** [comment]
@deiger I'm mapping it like this (https://github.com/deiger/Alarm/issues/12#issuecomment-771704470).  
I see the USB device with command ls -l /dev/serial/by-id and the previous version works perfectly with this (https://github.com/deiger/Alarm/issues/12#issuecomment-771704470)

**Comment by deiger** [comment]
According to the error message `/dev/serial` is a directory.
I suggest either use the first command I wrote (mounting to the inner directory), or change it to e.g. `/dev/serialport` (both in the mount command and in the options file.

**Comment by deiger** [comment]
Set `/opt/pima/options.json` to:
```json
{
  "log_level": "INFO",
  "login": "000000",
  "zones": "96",
  "mqtt_discovery_max_zone": 30,
  "mqtt_host": "XX.YY.ZZ.UU",
  "key": "28e6db84-44a4-410b-8f58-8c2275fdd396",
  "port": 7777,
  "mqtt_user": "homeassistant",
  "mqtt_pass": "password",
  "serialport": "/dev/serialport"
}
```
Then run:
```bash
sudo docker run -d --name pima --log-driver journald --restart always -v /opt/pima:/data --device /dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A50295BI-if00-port0:/dev/serialport deiger/pima:0.7.2.3
```

**Comment by Jens-Wymeersch** [comment]
@deiger Thank you for your quick reply and got a better answer still an error in the communication which i didn't have in the previous version.

`Traceback (most recent call last):


  File "pima_server.py", line 104, in run


    status = self._alarm.get_status()  # type: pima.Status


  File "/app/pima.py", line 193, in get_status


    raise Error('Invalid message {}.'.format(self._make_hex(response[2:3])))


pima.Error: Invalid message 0f.


I0207 17:10:13.211  pima_server.py:119] Trying to create the Alarm anew.


I0207 17:10:14.507  pima_server.py:152] Status: {'logged in': True, 'open zones': {3}, 'alarmed zones': set(), 'bypassed zones': set(), 'failed zones': set(), 'partitions': {1: 'disarm', 2: 'disarm', 3: 'disarm', 4: 'disarm', 5: 'disarm', 6: 'disarm', 7: 'disarm', 8: 'disarm', 9: 'disarm', 10: 'disarm', 11: 'disarm', 12: 'disarm', 13: 'disarm', 14: 'disarm', 15: 'disarm', 16: 'disarm'}, 'command ack': False}.


E0207 17:10:16.530  pima_server.py:111] Failed to get outputs status.


Traceback (most recent call last):


  File "pima_server.py", line 109, in run


    outputs = self._alarm.get_outputs()  # type: pima.Outputs


  File "/app/pima.py", line 274, in get_outputs


    raise Error('Invalid outputs response {}.'.format(self._make_hex(response)))


pima.Error: Invalid outputs response 08 0d 05 00 00 00 00 04 35.


E0207 17:10:26.340  pima_server.py:111] Failed to get outputs status.


Traceback (most recent call last):


  File "pima_server.py", line 109, in run


    outputs = self._alarm.get_outputs()  # type: pima.Outputs


  File "/app/pima.py", line 274, in get_outputs


    raise Error('Invalid outputs response {}.'.format(self._make_hex(response)))


pima.Error: Invalid outputs response 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 04 35 03.


E0207 17:10:27.343  pima_server.py:116] Exception raised by Alarm.`

**Comment by deiger** [comment]
Hmm..
The fetching the outputs sometimes fails, and the response is of the system status, even on retry. The good news is that the error is only cluttering the logs, but otherwise ignored. I'll change the code to avoid the clutter, as it is unhelpful.

**Comment by Jens-Wymeersch** [comment]
@deiger Dror, You are amazing. Thank you for your quick reply. Let me know when I need to do an update

**Comment by deiger** [comment]
You can, it's 0.7.2.4. The only difference is the log suppression.

**Comment by Jens-Wymeersch** [comment]
@deiger I'm going to leave it as is. Not really bothered with it as long as it doesn't interfere with the reading which you indicate it doesn't. I wonder what is the use case of the zones ? 
![image](https://user-images.githubusercontent.com/70786498/107159052-80b09880-6996-11eb-90bc-e67f7e51e15d.png)

Which information should I get and at what time ?

**Comment by deiger** [comment]
For each zone you get 2 indicator sensors: open (i.e. an open window or motion sensed), and alarming (i.e. the zone triggered an alarm).
As mentioned above, for the former you may wish to change the `data_class` for the open ones, to either `door`, `window` or `motion`, so you get the right symbols on open/close. Note that you'd better first rename the entities to something more human friendly.

**Comment by Jens-Wymeersch** [comment]
@deiger thanks. For those sensors, all are set as off and don’t change over time. Wired, wireless motion, door, window, etc. Will they only work in case of arm’ing of the system? 
for the device class, you indicate to change in customise.yaml ( I suppose this is in home assistant, right ?)

**Comment by l1c2d3** [comment]
> ```shell
> /opt/pima:/data --device /dev/serial
> ```

@deiger  Thank you for your answer. I customized already to their correspondent device class. 
I have another issue, after I arm the alarm (home or away), the alarm status changes to trigger in HA, but there is no zone triggered, what can it be causing it? In MQTT there is no alarmed zone.


@Jens-Wymeersch You can see the status changes from every sensor either when it's arm or disarmed.
The customizations, are on home assistant (configuration -> customizations), select the sensor you want to customize and the tap on select an attribute to override, then select device class, and select the class according to your sensor (motion, would be the appropriate for motion sensors)
If you have HA notifications already set, you can create automations to notify when an individual sensor or the panel is triggered.

**Comment by Jens-Wymeersch** [comment]
@l1c2d3 @deiger I have the alarm guy coming over tomorrow to understand which zones he actually used and change it accordingly. Anything I should specifically ask for besides
- Change or check : Arm Away does full arm, like a long press on 1. Arm Home arms home 1, like a long press on 4. The latter needs to be changed from 7 to 4
- looking at the zones and using the first 32
- understand which device is which zone

**Comment by deiger** [comment]
@l1c2d3 it turns out output 4 (ON/OFF Output) is always set when the alarm is armed. I've changed the code to only look at the siren outputs. Try version 0.7.2.5.

**Comment by deiger** [comment]
@Jens-Wymeersch the number 30 is arbitrary. The idea is that in order not to clutter HA, the server only generated the first x zones. Unfortunately I'm not aware of a way to ask the alarm if a zone is actually used.

**Comment by Jens-Wymeersch** [comment]
@deiger I can understand that the 30 is arbitrary. The technician is coming tomorrow morning. I'm going to set with him the first 30 zones and documenting them (so I can make the necessary changes in HA).

**Comment by Jens-Wymeersch** [comment]
@l1c2d3 can you please make a screen of your PIMA setup in Home Assistant. Thanks

**Comment by l1c2d3** [comment]
> @l1c2d3 it turns out output 4 (ON/OFF Output) is always set when the alarm is armed. I've changed the code to only look at the siren outputs. Try version 0.7.2.5.

Thanks. I will update and try it tonight

**Comment by l1c2d3** [comment]
> @l1c2d3 can you please make a screen of your PIMA setup in Home Assistant. Thanks

This is part of it, there are some more sensors. 
<img width="495" alt="Screen Shot 2021-02-08 at 12 58 31 p m" src="https://user-images.githubusercontent.com/28716784/107274790-20f5d400-6a0e-11eb-8f53-a747743d06ed.png">

**Comment by Jens-Wymeersch** [comment]
@l1c2d3 Looks nice but still also a lot of "off" I see. Why is that ? 
Secondly, did you try some automations with these sensors ? Is the state directly in sync ? Meaning if there is movement, is the state directly visible in HA ?

**Comment by l1c2d3** [comment]
> @l1c2d3 Looks nice but still also a lot of "off" I see. Why is that ?
> Secondly, did you try some automations with these sensors ? Is the state directly in sync ? Meaning if there is movement, is the state directly visible in HA ?

Yes, you can see the status changes in HA, there is a small delay for the door sensors, maybe 1 or 2 seconds, for the movement sensors I observed a much longer delay (I don't know why). 
I will start configuring automations for push notifications today.

**Comment by l1c2d3** [comment]
> @l1c2d3 are you using Dror’s docker image or the rpi py program. If the first, where do you do you make the changes ?

I'm not using the dock container, I'm running pima_server.py as a service, and I specified mqtt_discovery_max_zone as a parameter.
```
[Unit]
Description=PIMA alarm server
After=network.target

[Service]
ExecStart=/usr/bin/python3 -u pima_server.py --ssl_cert cert.pem --ssl_key key.pem --port 7777 --key my_key --login mycode -z 96  --mqtt_discovery_max_zone 20 --serialport /dev/serial0 --mqtt_host mqtt_ip --mqtt_user user:pass
WorkingDirectory=/usr/lib/pima
StandardOutput=inherit
StandardError=inherit
Restart=always

[Install]
WantedBy=multi-user.target
```
If you are running the docker container, you have to modify mqtt_discovery_max_zone in the options.json file, and restart the container. 
```
log_level: DEBUG
login: mycode
zones: 96
mqtt_discovery_max_zone: 20
mqtt_host: mqtt_ip
key: my_key
port: 7777

```

**Comment by Jens-Wymeersch** [comment]
@l1c2d3 Thank you for the advice, but it looks the wired zones are btw 1-10 and wireless 17-24. So won't gain that much. I'm now trying to figure out how to change the device_class. Anything you can share here ?

**Comment by deiger** [comment]
Then you can set mqtt_discovery_max_zone to 24, and disable zones 11-16 in HA.
For device_class, you have two options: Either editing customize.yaml, or using the UI under `Configuration -> Customization` and change the Device Class.

**Comment by Jens-Wymeersch** [comment]
@deiger Hi Dror, I've been testing the latest versions but had to go back to 0.7.1.6. 
The arming of the alarm doesn't work good any more. When you press alarm home, it doesn't update the state. Meaning you don't know if it is really arm'ed. When looking at the console, it actually arms after clicking twice. 
Besides the error you suppressed, I got the following in version 0.7.2.5

`  File "pima_server.py", line 104, in run,
    status = self._alarm.get_status()  # type: pima.Status,
  File "/app/pima.py", line 193, in get_status,
    raise Error('Invalid message {}.'.format(self._make_hex(response[2:3]))),
pima.Error: Invalid message 0f.,
I0209 09:47:46.126  pima_server.py:119] Trying to create the Alarm anew.,
I0209 09:47:47.438  pima_server.py:152] Status: {'logged in': True, 'open zones': set(), 'alarmed zones': set(), 'bypassed zones': set(), 'failed zones': set(), 'partitions': {1: 'disarm', 2: 'disarm', 3: 'disarm', 4: 'disarm', 5: 'disarm', 6: 'disarm', 7: 'disarm', 8: 'disarm', 9: 'disarm', 10: 'disarm', 11: 'disarm', 12: 'disarm', 13: 'disarm', 14: 'disarm', 15: 'disarm', 16: 'disarm'}, 'command ack': False}.,
I0209 09:47:59.246  pima_server.py:146] Status: {'logged in': True, 'open zones': set(), 'alarmed zones': set(), 'bypassed zones': set(), 'failed zones': set(), 'partitions': {1: 'disarm', 2: 'disarm', 3: 'disarm', 4: 'disarm', 5: 'disarm', 6: 'disarm', 7: 'disarm', 8: 'disarm', 9: 'disarm', 10: 'disarm', 11: 'disarm', 12: 'disarm', 13: 'disarm', 14: 'disarm', 15: 'disarm', 16: 'disarm'}, 'command ack': False, 'outputs': set()}.,
E0209 09:48:03.313  pima_server.py:116] Exception raised by Alarm.,
Traceback (most recent call last):,
  File "pima_server.py", line 104, in run,
    status = self._alarm.get_status()  # type: pima.Status,
  File "/app/pima.py", line 193, in get_status,
    raise Error('Invalid message {}.'.format(self._make_hex(response[2:3]))),
pima.Error: Invalid message 0f.,
I0209 09:48:03.314  pima_server.py:119] Trying to create the Alarm anew.,
I0209 09:48:04.582  pima_server.py:152] Status: {'logged in': True, 'open zones': set(), 'alarmed zones': set(), 'bypassed zones': set(), 'failed zones': set(), 'partitions': {1: 'disarm', 2: 'disarm', 3: 'disarm', 4: 'disarm', 5: 'disarm', 6: 'disarm', 7: 'disarm', 8: 'disarm', 9: 'disarm', 10: 'disarm', 11: 'disarm', 12: 'disarm', 13: 'disarm', 14: 'disarm', 15: 'disarm', 16: 'disarm'}, 'command ack': False}.,
E0209 09:48:08.664  pima_server.py:116] Exception raised by Alarm.,
Traceback (most recent call last):,
  File "pima_server.py", line 104, in run,
    status = self._alarm.get_status()  # type: pima.Status,
  File "/app/pima.py", line 193, in get_status,
    raise Error('Invalid message {}.'.format(self._make_hex(response[2:3]))),
pima.Error: Invalid message 0f.,
I0209 09:48:08.665  pima_server.py:119] Trying to create the Alarm anew.,
I0209 09:48:09.961  pima_server.py:152] Status: {'logged in': True, 'open zones': set(), 'alarmed zones': set(), 'bypassed zones': set(), 'failed zones': set(), 'partitions': {1: 'disarm', 2: 'disarm', 3: 'disarm', 4: 'disarm', 5: 'disarm', 6: 'disarm', 7: 'disarm', 8: 'disarm', 9: 'disarm', 10: 'disarm', 11: 'disarm', 12: 'disarm', 13: 'disarm', 14: 'disarm', 15: 'disarm', 16: 'disarm'}, 'command ack': False}.,
E0209 09:48:24.598  pima_server.py:116] Exception raised by Alarm.,
`

**Comment by Jens-Wymeersch** [comment]
@deiger How can I disable zones within the options.json file ?

**Comment by deiger** [comment]
You can disable them in HA.
AS for the error, it seems that get_status occasionally gets the outputs response. I'm sending a fix. It may be better though to make the whole thing asynchronous.

**Comment by Jens-Wymeersch** [comment]
Thanks @deiger for your reply. Sorry to ask but I wonder how to do disable them in HA? 
On the error, I didn’t have the error on 0.1.7.6
If it isn’t synchroon the sensors data becomes unuseful for any automations and arm’ing and unarm’ing should be synchroon anyhow.

**Comment by deiger** [comment]
If you click the unneeded sensor in `Configuration -> Entities`, there is an `Enable entity` toggle that you can switch to disable it.
The change in the recent version is the support for outputs, that is needed in order to alert that the alarm has triggered.

**Comment by Jens-Wymeersch** [comment]
@deiger Thank you for your guidance. I understand what you are saying, and I hope you fix it because this would be the cherry on the cake. 
As you have noticed, I'm ready and eager to test !

**Comment by deiger** [comment]
This is already in, check out 0.7.2.6.

**Comment by Jens-Wymeersch** [comment]
@deiger Thank you for the quick turn around. There are no error messages but the problem I've seen since 0.7.2.x is that the screen 

![image](https://user-images.githubusercontent.com/70786498/107488988-c8027900-6b90-11eb-8a50-6f882be30888.png)

when arming the alarm doesn't change to the following

![image](https://user-images.githubusercontent.com/70786498/107489448-4a8b3880-6b91-11eb-9d4f-2c45e6a3a0a1.png)

However, it arms the system on the console as I can see. In 0.7.1.6, the disarming requires multiple clicks. Maybe also something to look at.

**Comment by Jens-Wymeersch** [comment]
Hi @deiger I'm still having trouble arm'ing and disarming the alarm. Even with version 0.1.7.6, it isn't reliable. Here is an error I got lately 

`E0213 09:07:51.034  pima_server.py:87] Failed to create alarm object.
Traceback (most recent call last):
  File "pima_server.py", line 85, in __init__
    self._create_alarm()
  File "pima_server.py", line 140, in _create_alarm
    self._status = self._alarm.get_status()  # type: pima.Status
  File "/app/pima.py", line 177, in get_status
    response = self._read_message()
  File "/app/pima.py", line 269, in _read_message
    self._make_hex(data), self._crc(data), crc))
pima.Error: Invalid input on channel, CRC for 00 is 0, not 21858!`

Also sometimes after arm'ing and disarming, MQTT stops the connection. Can you please have a look ?

**Comment by Jens-Wymeersch** [comment]
@deiger I hope you have some time to have a look and help me out. 

I've redefined my strategy and would love the component to work directly with Home Assistant. I installed the component by adding https://github.com/deiger/Alarm in the Supervisor.

Here is my **config file**. I wasn't sure about the IP address of the MQTT broker. Should this be the HA IP address ?
```
zones 96
mqtt_discovery_max_zone 30
mqtt_host core-mosquitto
mqtt_port 1883
mqtt_topic pima_alarm
mqtt_user username
mqtt_pass userpw
mqtt_client_id pima
serialport /dev/ttyUSB0
pima_host HA IP
pima_port 0
key 28e6db84-44a4-410b-8f58-8c2275fdd396
port 7777
```

Initially I had a problem again with the serial port but by changing this to /dev/ssyUSB0 was able to overcome this but still have errors as you can see and can't get it to work.

`D0221 17:51:09.950  pima_server.py:70] Port: /dev/ttyUSB0.
D0221 17:51:10.813  pima.py:296] >>> 00 31 08
E0221 17:51:10.814  pima_server.py:87] Failed to create alarm object.
Traceback (most recent call last):
  File "pima_server.py", line 85, in __init__
    self._create_alarm()
  File "pima_server.py", line 151, in _create_alarm
    self._status = self._alarm.get_status()  # type: pima.Status
  File "/app/pima.py", line 179, in get_status
    response = self._read_message()
  File "/app/pima.py", line 300, in _read_message
    self._make_hex(data), self._crc(data), crc))
pima.Error: Invalid input on channel, CRC for 00 is 0, not 12552!`

**Comment by deiger** [comment]
`pima_host` is used in case you're connected via Ethernet rather than a serial port. You can just leave it empty.
Are you sure this is the correct USB port? The best way to use it is by running `ls -l /dev/serial/by-id`, and getting the right port ID, e.g. `/dev/serial/by-id/usb-1a86_USB2.0-Ser_-if00-port0`. That path won't change even if new USB peripherals are added.

**Comment by Jens-Wymeersch** [comment]
@deiger I never changed the connection, just changed the SD card. That said, I moved back to my initial idea of keeping the product in a separate docker and connect it to MQTT. The problem I'm facing is that every day, I need to restart the docker as somehow (no logs), it stops working in 0.7.1.6 and in 0.7.2.6 I can't arm the system somehow. I would prefer moving to the newer version and not having this issue. Can you please have a look ?

**Comment by Jens-Wymeersch** [comment]
@deiger Hi Dror, it's been a while that I pick this up again because I wanted to move to the newer version, change the RPI from 4 to 3. I get the same error for 0.1.6.7 as 0.2.6.7. Would it be possible to have a look again ?

`E0731 13:40:41.932  pima_server.py:87] Failed to create alarm object.
Traceback (most recent call last):
  File "pima_server.py", line 85, in __init__
    self._create_alarm()
  File "pima_server.py", line 151, in _create_alarm
    self._status = self._alarm.get_status()  # type: pima.Status
  File "/app/pima.py", line 179, in get_status
    response = self._read_message()
  File "/app/pima.py", line 300, in _read_message
    self._make_hex(data), self._crc(data), crc))
pima.Error: Invalid input on channel, CRC for 00 is 0, not 8!`

**Comment by Jens-Wymeersch** [comment]
@deiger Adding the latest error messages. Half of the time, I'm not  able to reliably arm/disarm the alarm. Can you please help?
@l1c2d3 Does it work with you now ? How did you do your setup ?

**Comment by Jens-Wymeersch** [comment]
This is the latest error while trying to activate the alarm 


E0801 16:26:10.756  pima_server.py:109] Exception raised by Alarm.

Traceback (most recent call last):

  File "pima_server.py", line 102, in run

    status = self._alarm.get_status()  # type: pima.Status

  File "/app/pima.py", line 177, in get_status

    response = self._read_message()

  File "/app/pima.py", line 264, in _read_message

    self._make_hex(data), length + 3))

pima.Error: Not enough data in channel: 62 0d 05 01 02 00 00 00 00 00 00 00 00 00 00 00 00 00 should have 101 bytes.

I0801 16:26:10.758  pima_server.py:112] Trying to create the Alarm anew.

**Comment by l1c2d3** [comment]
> @l1c2d3 Does it work with you now ? How did you do your setup ?
It hangs from time to time, mostly when disarming (either armed or triggered), although it sometimes gets the command sent before hanging, sometimes it doesn't.
Everything comes back to normal, after restarting the service.

**Comment by l1c2d3** [comment]
> > > @l1c2d3 Does it work with you now ? How did you do your setup ?
> > 
> > 
> > It hangs from time to time, mostly when disarming (either armed or triggered), although it sometimes gets the command sent before hanging, sometimes it doesn't.
> > Everything comes back to normal, after restarting the service.
> 
> Hello,
> 
> Are you using the install directly on the RPI, the docker or HA version Dror made ?
> 
> Regards,
> Jens

The alarm is connected to a Raspberry Pi Zero W. 
It uses Dror's script as a service (I followed the procedure in the project documentation),  which sends the information via MQTT to my HA instance, where I also have the Pima Alarm add-on for HA configured.
